iml-wire-types/src/lib.rs

```diff
@@ -1235,14 +1235,30 @@ pub struct Substitution {
     pub resource_uri: String,
 }
 
-#[derive(serde::Deserialize, serde::Serialize, Clone, Debug)]
+#[derive(serde::Deserialize, serde::Serialize, Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "postgres-interop", derive(sqlx::Type))]
+#[cfg_attr(feature = "graphql", derive(juniper::GraphQLEnum))]
 #[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+#[repr(i16)]
 pub enum MessageClass {
-    Normal,
-    Lustre,
-    LustreError,
-    Copytool,
-    CopytoolError,
+    Normal = 0,
+    Lustre = 1,
+    LustreError = 2,
+    Copytool = 3,
+    CopytoolError = 4,
+}
+
+impl From<i16> for MessageClass {
```

Comment:
This needs to be a `TryFrom`.

Any panic is an unchecked error that won't be handled in normal control flow.

In addition, we can't guarantee future usages of the conversion even if this one is deemed safe.

Reply:
Done